### PR TITLE
update operator status when the operand is unmanaged

### PR DESCRIPTION
**Pull request by bparees**
https://bugzilla.redhat.com/show_bug.cgi?id=1667376

### CODE REVIEW

**Comment by bparees** [comment]
> @bparees do we want all our team's operators to follow this pattern wrt Unmanaged and the three clusteroperator status conditions ?

@gabemontero it's as reasonable as anything else i've come up with...  the main point of debate was whether available should be true/false(depending on the operand) or unknown.  It is possible(at least in some cases) for an operator to understand if its operand is available or not, even when it's not actively managing that operand.  I chose not to implement that logic at least for now (ie pending pushback).

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -80,10 +83,36 @@ func (c OpenShiftControllerManagerOperator) sync() error {
...
 	switch operatorConfig.Spec.ManagementState {
-	case operatorsv1.Unmanaged:
+	case operatorv1.Unmanaged:
+		// manage status
+		originalOperatorConfig := operatorConfig.DeepCopy()
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeAvailable,
+			Status:  operatorv1.ConditionUnknown,
+			Reason:  "Unmanaged",
+			Message: "the controller manager is in an unmanaged state, therefore its availability is unknown.",
```
Comment (coreydaley) [inline pkg/operator/operator.go]: s/its/it's

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -80,10 +83,36 @@ func (c OpenShiftControllerManagerOperator) sync() error {
...
+		})
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeProgressing,
+			Status:  operatorv1.ConditionFalse,
```
Comment (coreydaley) [inline pkg/operator/operator.go]: Should this also be unknown instead of false?

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -80,10 +83,36 @@ func (c OpenShiftControllerManagerOperator) sync() error {
...
+			Reason:  "Unmanaged",
+			Message: "the controller manager is in an unmanaged state, therefore no changes are being applied.",
+		})
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeFailing,
+			Status:  operatorv1.ConditionFalse,
```
Comment (coreydaley) [inline pkg/operator/operator.go]: Should this also be unknown?

**Pull request merged**